x/leverage/types/token.go

```diff
@@ -96,6 +96,10 @@ func (t Token) Validate() error {
 		return sdkerrors.ErrInvalidRequest.Wrap("Token.MinCollateralLiquidity be between 0 and 1")
 	}
 
+	if !t.MaxSupply.IsPositive() {
+		return sdkerrors.ErrInvalidRequest.Wrap("Token.MaxSupply must be positive")
+	}
```

Comment:
This rule would make for a more clear token registry.

```suggestion
	if !t.EnableMsgSupply && MaxSupply.IsPositive() {
		return sdkerrors.ErrInvalidRequest.Wrap("MaxSupply must be zero when EnableMsgSupply is false")
	}
	if t.MaxSupply.IsNegative() {
		return sdkerrors.ErrInvalidRequest.Wrap("Token.MaxSupply must not be negative")
	}
```

That way, any token with supply disabled has an accurate `MaxSupply` of `0` (I fear that a nonzero max supply, when supply is disabled, would be misleading).

Reply:
OK, however we should use `MaxSupply.IsZero()` instead of `MaxSupply.IsPositive()`

Reply:
Also it will change the proto comment - currently I set in proto that zero is not a valid number. Will update both.